doc/classes/NetworkedMultiplayerCustom.xml

```diff
@@ -32,8 +32,9 @@
 			<argument index="0" name="connection_status" type="int" enum="NetworkedMultiplayerPeer.ConnectionStatus" />
 			<description>
 				Set the state of the connection. See [enum NetworkedMultiplayerPeer.ConnectionStatus].
-				This will emit the [signal NetworkedMultiplayerPeer.connection_succeeded], [signal NetworkedMultiplayerPeer.connection_failed] or [signal NetworkedMultiplayerPeer.server_disconnected] signals depending on the status and if the peer has the unique network id of [code]1[/code].
+				This will emit the [signal NetworkedMultiplayerPeer.connection_succeeded] signal when set to [constant NetworkedMultiplayerPeer.CONNECTION_CONNECTED].
 				You can only change to [constant NetworkedMultiplayerPeer.CONNECTION_CONNECTING] from [constant NetworkedMultiplayerPeer.CONNECTION_DISCONNECTED] and to [constant NetworkedMultiplayerPeer.CONNECTION_CONNECTED] from [constant NetworkedMultiplayerPeer.CONNECTION_CONNECTING].
+				Setting this to [constant NetworkedMultiplayerPeer.CONNECTION_DISCONNECTED] will emit the [signal NetworkedMultiplayerPeer.connection_failed] if it was previously set to [constant NetworkedMultiplayerPeer.CONNECTION_CONNECTING] or [signal NetworkedMultiplayerPeer.server_disconnected] if previously [constant NetworkedMultiplayerPeer.CONNECTION_CONNECTED]. Note that this is only true if the peer is not in server mode (has the unique network id of [code]1[/code]), otherwise the connection status will be set to [constant NetworkedMultiplayerPeer.CONNECTION_DISCONNECTED] without emitting any signals.
```

Comment:
This much detail in the docs for `set_connection_status()` is quite difficult to follow. I think the main thing that developers need to know about the signals, is that `connection_succeeded`, `connection_failed` and `server_disconnected` will get emitted for them (whereas they need to emit `peer_connected` and `peer_disconnected` themselves, although I'm not sure where to put that note).

This all might be better explained by including a short example implementation?

Reply:
I agree but I think that the problem I'm having is that this bit has some [control coupling](https://en.wikipedia.org/wiki/Coupling_(computer_programming)#Object-oriented_programming:~:text=tools%20and%20devices.-,Control%20coupling,-Control%20coupling%20is) going on. Extra documentation doesn't solve this issue but it alleviates the user experience somewhat. Like you said, knowing the signals get emitted is the most important bit. I felt in the dark as a user since this is a "custom" implementation knowing when and how those signals actually were emitted. It's not obvious I think from the signal docs since they rely on the implementations to call them on whether or not they're handled here.

Reply:
Well, the goal here is that the developer just calls `set_connection_status()` and the right signals get emitted (in an effort to _reduce_ the number of things the developer needs to consider).

I haven't tried to write it (so it might not be possible to do briefly), but I think a code snippet in the description might be the clearest way to show developers what they need to do.

Reply:
This isn't as brief as I'd like it, but this is possibly a first pass at a code snippet:

```gdscript
var custom_peer := NetworkedMultiplayerCustom.new()
var real_multiplayer_client

func join_multiplayer():
	custom_peer.set_connection_status(NetworkedMultiplayerPeer.CONNECTION_CONNECTING)
	get_tree().network_peer = custom_peer

	# Start connecting via real transport (ex. Steam peer-to-peer, PlayFab, etc).
	real_multiplayer_client = _connect_to_real_multiplayer()
	real_multiplayer_client.connect("connection_succeeded", self, "_on_real_multiplayer_connection_succeeded")
	real_multiplayer_client.connect("connection_failed", self, "_on_real_multiplayer_connection_failed")
	real_multiplayer_client.connect("peer_connected", self, "_on_real_multiplayer_peer_connected")
	real_multiplayer_client.connect("peer_disconnected", self, "_on_real_multiplayer_peer_disconnected")
	real_multiplayer_client.connect("recieved_message", self, "_on_real_multiplayer_received_message")

	# Intercept any RPC's sent locally.
	custom_peer.connect("packet_generated", self, "_on_custom_peer_packet_generated")

func _on_real_multiplayer_connection_succeeded():
	# Once connection is formed, assign a peer id somehow. If the real transport
	# doesn't have a compatible id, you'll need to generate your own, and ensure
	# that all peers use the same mapping.
	var my_id = real_multiplayer_client.get_my_id()
	custom_peer.initialize(my_id)
	custom_peer.set_connection_status(NetworkedMultiplayerCustom.CONNECTION_CONNECTED)

func _on_real_multiplayer_connection_failed():
	custom_peer.set_connection_status(NetworkedMultiplayerCustom.CONNECTION_DISCONNECTED)

func _on_real_multiplayer_peer_connected(peer_id):
	# Let Godot know a peer connected. Again, you may need to map between Godot ids
	# and the ids used by the real multiplayer client.
	custom_peer.emit_signal("peer_connected", peer_id)

func _on_real_multiplayer_peer_disconnected(peer_id):
	# Same notes as above.
	custom_peer.emit_signal("peer_disconnected", peer_id)

func _on_custom_peer_packet_generated(peer_id, buffer, transfer_mode):
	# Send any packets generated locally (ex RPCs) to the requested peer
	# using the real multiplayer client.
	if transfer_mode == NetworkedMultiplayerPeer.TRANSFER_MODE_RELIABLE:
		real_multiplayer_client.send_reliable(peer_id, {
			'from_peer_id': custom_peer.get_unique_id(),
			'data': buffer,
		})
	else:
		real_multiplayer_client.send_unreliable(peer_id, {
			'from_peer_id': custom_peer.get_unique_id(),
			'data': buffer,
		})

func _on_real_multiplayer_received_message(message):
	# Take any packets we recieved from other peers, and deliver them locally.
	custom_peer.deliver_packet(message['data'], message['from_peer_id'])
```